src/Catalyst.Core.Modules.Web3/ApiModule.cs

```diff
@@ -58,23 +62,26 @@ protected override void Load(ContainerBuilder builder)
             var buildPath = Path.GetDirectoryName(executingAssembly);
             var webDirectory = Directory.CreateDirectory(Path.Combine(buildPath, "wwwroot"));
 
-            var host = WebHost.CreateDefaultBuilder()
-               .ConfigureServices(serviceCollection => ConfigureServices(serviceCollection, builder))
-               .Configure(Configure)
-               .UseUrls(_apiBindingAddress)
-               .UseWebRoot(webDirectory.FullName)
-               .UseSerilog()
-               .Build();
-
-            builder.RegisterInstance(host).As<IWebHost>();
-
             async void BuildCallback(IContainer container)
             {
                 _container = container;
                 var logger = _container.Resolve<ILogger>();
 
                 try
                 {
+                    var host = Host.CreateDefaultBuilder()
+                        .UseServiceProviderFactory(new AutofacServiceProviderFactory())
```

Comment:
not sure why we're doing it this way, you must use the same container otherwise it provides two instances of mempool, are we not able to pass in the existing container like before?

Reply:
No its using the same Mempool, I have tested, researched. It is a new breaking change in .net 3.0 I believe you cannot use the old way we were using hence why it broke in the first place.